File: backend/capops_api/views/elastic.py
"""
Author: Deric Le
Description: Define functions to be used for Elastic API
"""

import random 

from django.shortcuts import render
from rest_framework.decorators import api_view
from rest_framework.response import Response

@api_view(['GET'])
def getCriticals(request):
    criticals = random.randint(1, 10)
    return Response({'data': criticals})

@api_view(['GET'])
def getHighs(request):
    highs = random.randint(1, 100)
    return Response({'data': highs})

File: frontend/src/components/Carousel/items/elastic.js
/**
 * Author: Deric Le
 * Description: Handles Elastic API
 */

const FetchEL = async () => {
    try {
        const fetchCriticals = await fetch('http://localhost:8000/api/EL/detections/criticals');
        if (!fetchCriticals.ok) { throw new Error('Failed to fetch criticals from EL'); }

        const fetchHighs = await fetch('http://localhost:8000/api/EL/detections/highs');
        if (!fetchHighs.ok) { throw new Error('Failed to fetch criticals from EL'); }

        const criticals = await fetchCriticals.json();
        const highs = await fetchHighs.json();

        return {criticals: criticals, highs: highs};

    } catch (error) { 
        console.error('Error fetching data:', error);
        return null;
    }
    
};

export default FetchEL;
